src/main.cpp

```diff
@@ -588,7 +588,10 @@ int main(int argc, char **argv) {
                 if (strcmp(arg, "--output") == 0) {
                     out_file = argv[i];
                 } else if (strcmp(arg, "--output-h") == 0) {
-                    out_file_h = argv[i];
+                    assert(out_type != OutTypeUnknown);
+                    if (out_type == OutTypeObj || out_type == OutTypeLib) {
```

Comment:
I think you should probably only collect the value here, and do this logic after iterating all the args. Otherwise it would make the argument order important.

Reply:
Indeed.